### DISCUSSION TITLE
Can a personal account have access to repos from my work account (which have private repos) ?

### DISCUSSION THREAD

**Discussion by devluist** [community/community#49912] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I want to buy a license of copilot for a year, but not sure which account should I use...

I wish to have the license on my personal account so that I can work on personal demos/learning/courses, but at the same time I am worried that since my work account (another GitHub account with it is own email for the company) has private repos, I am not sure if that will limit the access to those repos if I set the license on my personal account.

So the question is, Can I access private repos from another email account (that I have access to) using Individual license copilot?

Or should I set the license on my work account and forget about using copilot for personal stuff or local learning/examples?

**Comment by Anmol-Baranwal** [comment, marked as answer]
If you want to use GitHub Copilot with your personal account and access private repositories associated with your work account, you can use the personal access token from your work account to authenticate the requests made by GitHub Copilot.

To do so, you'll need to create a personal access token on your work account with the appropriate scopes to access the private repositories you need. Then, you can configure GitHub Copilot to use this token for authentication by setting the `GITHUB_TOKEN` environment variable to the token value.

Keep in mind that when you use the personal access token from your work account to access private repositories, GitHub Copilot will use that token to make requests on your behalf, so any actions you take will be attributed to your work account. Also, make sure that you comply with your company's policies regarding the use of their resources for personal projects.

Alternatively, if you prefer to keep your personal and work accounts separate, you could consider purchasing a separate license for your work account to use GitHub Copilot for work-related projects. This way, you can keep your personal and work projects separate while still being able to access private repositories associated with your work account.

> I hope this might resolve your issue.

**Reply by adrianschmidt** [reply]
I'm in a similar situation as devluist, but with an important difference. I do _not_ have separate accounts, but use a single GitHub account for both personal and work access. I would like to try Copilot with an Individual subscription, but since our company policy so far is that employees are not allowed to give GPT or similar AI access to code in private repos, I want to know beforehand if I can control which repos Copilot get access to?

Company repos are held under the company's GitHub Organization. However, the same Organization also has open source repos, which I am allowed to, and would like to, use Copilot on. Can I allow access only to such repos?

**Reply by Anmol-Baranwal** [reply]
Yes, you can control which repositories Copilot can access by granting access only to specific repositories. When you subscribe to Copilot as an individual, you'll be asked to authorize the application to access your GitHub account. During this process, you can choose which repositories Copilot has access to by selecting the appropriate permissions.

To do this, you can go to the "Settings" section of your GitHub account, click on "Applications", and then click on "Authorized GitHub Apps". From there, you can select Copilot and then choose which repositories you want to grant it access to.

> So, you can grant Copilot access to only those specific repositories.

**Reply by adrianschmidt** [reply]
Hmm… I can't actually see Copilot in that section. I have two other apps though.

I do have a in the Settings sidebar, but if I click that I only get options whether to allow suggestions matching public code, and whether GitHub can use my code snippets for product improvements. Nothing about access to different repos.

I guess I'll simply have to only enable Copilot in VS Code while I only have the open source repos open?

**Reply by mchaliadzinau** [reply]
Also interested in answer on @adrianschmidt question.

**Comment by josedera23** [comment]
I have an organization on github with 10 seats, and pay for the GithubTeam. My question is..... Can my team use github copilot individual, and pay it individual? Or exist some Compliance for that?. Thanks
